solr/core/src/test/org/apache/solr/cloud/TestConfigSetsAPI.java

```diff
@@ -198,54 +198,84 @@ public void testCreate() throws Exception {
   @Test
   public void testCreateWithTrust() throws Exception {
     String configsetName = "regular";
-    String configsetSuffix = "testCreateWithTrust";
-    String configsetSuffix2 = "testCreateWithTrust2";
-    uploadConfigSetWithAssertions(configsetName, configsetSuffix, "solr");
-    uploadConfigSetWithAssertions(configsetName, configsetSuffix2, null);
+    String trustedConfigsetSuffix = "testCreateWithTrustTrusted";
+    String untrustedConfigsetSuffix = "testCreateWithTrustUntrusted";
+    uploadConfigSetWithAssertions(configsetName, trustedConfigsetSuffix, "solr");
+    uploadConfigSetWithAssertions(configsetName, untrustedConfigsetSuffix, null);
     try (SolrZkClient zkClient =
         new SolrZkClient(
             cluster.getZkServer().getZkAddress(), AbstractZkTestCase.TIMEOUT, 45000, null)) {
-      assertTrue(isTrusted(zkClient, configsetName, configsetSuffix));
-      assertFalse(isTrusted(zkClient, configsetName, configsetSuffix2));
+      assertTrue(isTrusted(zkClient, configsetName, trustedConfigsetSuffix));
+      assertFalse(isTrusted(zkClient, configsetName, untrustedConfigsetSuffix));
+      // base trusted -> untrusted
+      try {
+        ignoreException("unauthenticated request");
+        // trusted -> untrusted
+        createConfigSet(
+            null, // base is default trusted
+            "abc",
+            Collections.emptyMap(),
+            cluster.getSolrClient(),
+            null); // without username is untrusted
+        fail("Expecting exception");
+      } catch (SolrException e) {
+        assertEquals(SolrException.ErrorCode.UNAUTHORIZED.code, e.code());
```

Comment:
Do we expect an exception always? Should use assertThrows if so.

Reply:
yea... I just copied how the test was written previously. I'll clean it up.

Reply:
Fixed in 2011fb0c51476478322e41a4f2b5259f44ea6662